src/main.cpp

```diff
@@ -769,6 +773,10 @@ int main(int argc, char **argv)
 	if ( !initOutput( g_nPreferredOutputWidth, g_nPreferredOutputHeight, g_nNestedRefresh ) )
 	{
 		fprintf( stderr, "Failed to initialize output\n" );
+		if ( BIsSDLSession() ) {
+			g_sdlLatch.store(1, std::memory_order_release);
```

Comment:
Do we really need to do all this on the failure paths? Maybe we should just `exit(1)`.

Reply:
> Do we really need to do all this on the failure paths? Maybe we should just `exit(1)`.


I've now refactored the code to register a cleanup function and call `exit(1)` on the applicable failure paths

Reply:
I don't even think we need the atexit stuff do we? The kernel will clean up everything there.

Reply:
> I don't even think we need the atexit stuff do we? The kernel will clean up everything there.


Good point
I've gotten rid of the atexit stuff now